Enhance CreateEffectWarhead

penev92 opened the pull request:
This adds some randomness to the audio and visual effects of `CreateEffectWarhead` as requested by @DoGyAUT.

Also closes #10629.

penev92 committed:
Add support for multiple explosions to CreateEffectWarhead

Have the engine pick one explosion at random from the provided list.

penev92 committed:
Add support for multiple impact sounds to CreateEffectWarhead

Have the engine pick one impact sound at random from the provided list.

Phrohdoh commented on OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs:
```diff
@@ -17,17 +17,17 @@ namespace OpenRA.Mods.Common.Warheads
-		[Desc("Explosion effect to use.")]
-		public readonly string Explosion = null;
+		[Desc("List of explosion effects that can be used.")]
+		public readonly List<string> Explosion = new List<string>();
```
This now needs to be `Explosions`.

pchote replied:
Lets make these `string[]`s too, please.

Phrohdoh commented on OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs:
```diff
@@ -17,17 +17,17 @@ namespace OpenRA.Mods.Common.Warheads
-		[Desc("Sound to play on impact.")]
-		public readonly string ImpactSound = null;
+		[Desc("List of sounds that can be played on impact.")]
+		public readonly List<string> ImpactSound = new List<string>();
```
Likewise `ImpactSounds`.

reaperrr replied:
And don't forget the upgrade rules.

penev92 committed:
Clean up some dead YAML from Fort Lonestar and Drop Zone W weapons definitions

There haven't been WaterExplosion and WaterImpactSound properties on CreateEffectWarhead for quite a while.

penev92 commented:
Fixed `--check-yaml`, rebased and applied requested changes.
Also added another commit to clean up some of the Fort Lonestar and Drop Zone W weapon definitions.

RoosterDragon commented:
:+1: Looks good to me.

obrakmann commented:
:+1:

obrakmann commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/63626aadca490f8425cfaf74be010f391239265f...757e6d5a04465261a48c5129a964e837b33114eb)
